sycl/test/esimd/enums.cpp

```diff
@@ -1,4 +1,4 @@
-// RUN: %clangxx -fsycl -fsyntax-only -Xclang -verify %s
+// RUN: %clangxx -fsycl -fsycl-device-only -fsyntax-only -Xclang -verify %s
```

Comment:
We still want to check host compilation of ESIMD APIS.
I found some explanation here https://github.com/intel/llvm/pull/3777/commits/7669de4d968bcecca1bbb563f671e68771a090f2:

> The reason for the change is that "-verify" mode does not work correctly
with new integration footer approach, because it seems to ignore line
markers in the input file.

But can you please elaborate on it? To me it seems that we silently hide some problem in host compilation.

Reply:
> But can you please elaborate on it?

Sure, I plan to outline those diagnostics tests changes into a separate PR with more detailed commit message - stay tuned :)

> To me it seems that we silently hide some problem in host compilation.

Hopefully, we don't have different API endpoints for host and device APIs. BTW, is ESIMD classes like `simd` are expected to work in host part of the program?

Reply:
> Hopefully, we don't have different API endpoints for host and device APIs.

Actually, for ESIMD, we do :) . device APIs are lowered into GenX intrinsics, while for host we have an impl using simple loops. That's why we still want to check host compilation, because it's two different paths.

> BTW, is ESIMD classes like simd are expected to work in host part of the program?

No, but they are expected to work in the device part of the program executed on the host.

Reply:
@DenisBakhvalov, I've created separate PR for those test changes and copied our discussion here: [#3956 (comment)](https://github.com/intel/llvm/pull/3956/files#r654581863)